lib/integrations/awsoidc/deployservice_update.go

```diff
@@ -0,0 +1,220 @@
+	ecsTypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/modules"
+)
+
+// waitDuration specifies the amount of time to wait for a service to become healthy after an update.
+const waitDuration = time.Minute * 5
+
+// UpdateDeployServiceAgents updates the deploy service agents with the specified teleportVersionTag.
+func UpdateDeployServiceAgents(ctx context.Context, clt DeployServiceClient, teleportClusterName, teleportVersionTag string, ownershipTags AWSTags) error {
+	teleportFlavor := teleportOSS
+	if modules.GetModules().BuildType() == modules.BuildEnterprise {
+		teleportFlavor = teleportEnt
+	}
+	teleportImage := fmt.Sprintf("public.ecr.aws/gravitational/%s-distroless:%s", teleportFlavor, teleportVersionTag)
+
+	if err := updateDeployServiceAgent(ctx, clt, teleportClusterName, teleportImage, ownershipTags); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func updateDeployServiceAgent(ctx context.Context, clt DeployServiceClient, teleportClusterName, teleportImage string, ownershipTags AWSTags) error {
+	service, err := getManagedService(ctx, clt, teleportClusterName, ownershipTags)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	taskDefinition, err := getManagedTaskDefinition(ctx, clt, aws.ToString(service.TaskDefinition), ownershipTags)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	currentTeleportImage, err := getTaskDefinitionTeleportImage(taskDefinition)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if currentTeleportImage == teleportImage {
+		return nil
+	}
+
+	registerTaskDefinitionIn, err := generateTaskDefinitionWithImage(taskDefinition, teleportImage, ownershipTags.ToECSTags())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	registerTaskDefinitionOut, err := clt.RegisterTaskDefinition(ctx, registerTaskDefinitionIn)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Update service with new task definition
+	_, err = updateServiceOrRollback(ctx, clt, service, registerTaskDefinitionOut.TaskDefinition)
+	if err != nil {
+		// If update failed, then rollback task definition
+		_, rollbackErr := clt.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+			TaskDefinition: registerTaskDefinitionOut.TaskDefinition.TaskDefinitionArn,
+		})
+		if rollbackErr != nil {
+			return trace.Wrap(err, "failed to rollback task definition: %v", rollbackErr)
+		}
+		return trace.Wrap(err)
+	}
+
+	// Attempt to deregister previous task definition but ignore error on failure
+	clt.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+		TaskDefinition: taskDefinition.TaskDefinitionArn,
+	})
+	return nil
+}
+
+func getManagedService(ctx context.Context, clt DeployServiceClient, teleportClusterName string, ownershipTags AWSTags) (*ecsTypes.Service, error) {
+	ecsClusterName := fmt.Sprintf("%s-teleport", normalizeECSResourceName(teleportClusterName))
```

Comment:
Is there a function/constant for this? Feels like this is used in more than a single place.